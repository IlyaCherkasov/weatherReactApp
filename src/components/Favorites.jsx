import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import FavTown from './FavTown';
import * as favoritesSelector from '../favorites/selectors';
import { getFavorites, addFavorite } from '../favorites/actions';

const Header = styled.div`
  display: flex;
  justify-content: space-between;
  margin-top: 40px;
`;

const Text = styled.div`
  font-size: 30pt;
  font-weight: 500;
  color: #85898C;
`;

const NewCityInput = styled.input`
  border: none;
  background-color: #F1F1F1;
  color: #8B9396;
  padding: 0 12px;
  width: 400px;
  margin-right: 20px;
  border-radius: 2px;
  height: 100%;
`;

const AddButton = styled.button`
  border: none;
  background-color: #708189;
  color: #ffffff;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  font-size: 25pt;
`;

const Towns = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
  margin-top: 40px;
  flex-wrap: wrap;
`;

class Favorites extends React.Component {
  componentDidMount() {
    this.props.getFavorites();
  }

  handleAddClick (e) {
    e.preventDefault();
    if (e.target[0].value !== '') {
      this.props.addFavorite(e.target[0].value);
      e.target[0].value = '';
    }
  }

  render() {
    const { favorites } = this.props;
    return (
      <>
        <Header>
          <Text>Избранное</Text>
          <form style={{display: 'flex'}} onSubmit={e => this.handleAddClick(e)} data-testid="form">
            <NewCityInput placeholder="Добавить новый город"/>
            <AddButton type="submit">+</AddButton>
          </form>
        </Header>
        <Towns>
          {favorites  && favorites.map(townName => (
            <FavTown
              key={townName}
              name={townName}
            />
          ))}
        </Towns>
      </>
    );
  }
}

const mapStateToProps = state => ({
  favorites: favoritesSelector.getFavorites(state),
});

const mapDispatchToProps = {
  getFavorites,
  addFavorite,
};

export default connect(mapStateToProps, mapDispatchToProps)(Favorites);
