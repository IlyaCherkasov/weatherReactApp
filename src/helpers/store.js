import thunkMiddleware from 'redux-thunk';
import {applyMiddleware, compose, createStore} from 'redux';
import rootReducer from '../store/rootReducer';

const middlewares = [thunkMiddleware];

export const store = compose(applyMiddleware(...middlewares))(createStore)(rootReducer);