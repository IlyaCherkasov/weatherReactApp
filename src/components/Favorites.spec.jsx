import React from 'react';
import 'jest-styled-components';
import { fireEvent } from '@testing-library/react';
import { renderWithRedux as render } from "../helpers/renderWith";
import Favorites from './Favorites';

describe('Favorites component', () => {
  it('should render correctly without children', () => {
    const { container, getByPlaceholderText, getByTestId } = render(<Favorites />, {});
    expect(container.firstChild).toMatchSnapshot();
    fireEvent.change(getByPlaceholderText('Добавить новый город'), { target: { value: 'Moscow' }});
    expect(getByPlaceholderText('Добавить новый город').value).toBe('Moscow');
    // submit - synthetic event that options should be written
    fireEvent.submit(getByTestId('form'), { target: [{ value: 'Moscow' }] });
  });

  it('should render correctly with children', () => {
    const { container } = render(<Favorites />, { initialState: {
      favorites: { favorites: ['Moscow'] },
      weather: { namedWeather: {
          Moscow:  { weather: {
              main: { temp: 283, pressure: 10000, humidity: 10 },
              weather: [{ icon: 10 }],
              wind: { speed: 10, deg: 10 },
              clouds: { all: 10 },
              coord: { lon: 10, lat: 10 },
            } }  } }
      } });
    expect(container).toMatchSnapshot();
  });
});