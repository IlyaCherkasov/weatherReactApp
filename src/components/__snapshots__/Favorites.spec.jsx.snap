// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component should render correctly with children 1`] = `
.c11 {
  width: 100%;
  text-align: center;
  margin-top: 20px;
}

.c12 {
  font-size: 25px;
  text-align: center;
  color: #6F8185;
}

.c13 {
  width: 250px;
  height: 80px;
  margin-top: 20px;
  background-color: #C9D2D7;
  border: none;
}

.c13 svg {
  width: 50px;
  height: 50px;
  fill: #6F8185;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 46px;
}

.c5 {
  width: 45%;
  margin-bottom: 20px;
}

.c7 {
  font-size: 26pt;
  font-weight: 700;
  color: #71848B;
}

.c8 {
  font-size: 26pt;
  color: #71848B;
  display: none;
}

.c9 {
  width: 50px;
  height: 40px;
  float: left;
  border: 3px solid #71848B;
  text-align: center;
  color: #6E8084;
  font-size: 10pt;
  display: none;
}

.c9 img {
  width: 50px;
  height: 40px;
}

.c10 {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
  border: none;
  background-color: #708189;
  color: #ffffff;
  border-radius: 50%;
  padding: 0 14px;
  font-size: 25pt;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 40px;
}

.c1 {
  font-size: 30pt;
  font-weight: 500;
  color: #85898C;
}

.c2 {
  border: none;
  background-color: #F1F1F1;
  color: #8B9396;
  padding: 0 12px;
  width: 400px;
  margin-right: 20px;
  border-radius: 2px;
  height: 100%;
}

.c3 {
  border: none;
  background-color: #708189;
  color: #ffffff;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  font-size: 25pt;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 40px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Избранное
    </div>
    <form
      data-testid="form"
      style="display: flex;"
    >
      <input
        class="c2"
        placeholder="Добавить новый город"
      />
      <button
        class="c3"
        type="submit"
      >
        +
      </button>
    </form>
  </div>
  <div
    class="c4"
  >
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          Moscow
        </div>
        <div
          class="c8"
        >
          10
          °C
        </div>
        <div
          class="c9"
        >
          <img
            alt="icon"
            src="https://openweathermap.org/img/w/10.png"
          />
        </div>
        <button
          class="c10"
        >
          +
        </button>
      </div>
      <div
        class="c11"
      >
        <div
          class="c12"
        >
          Подождите, данные загружаются
        </div>
        <button
          class="c13"
        >
          <svg>
            Refresh.svg
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Favorites component should render correctly without children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 40px;
}

.c1 {
  font-size: 30pt;
  font-weight: 500;
  color: #85898C;
}

.c2 {
  border: none;
  background-color: #F1F1F1;
  color: #8B9396;
  padding: 0 12px;
  width: 400px;
  margin-right: 20px;
  border-radius: 2px;
  height: 100%;
}

.c3 {
  border: none;
  background-color: #708189;
  color: #ffffff;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  font-size: 25pt;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Избранное
  </div>
  <form
    data-testid="form"
    style="display: flex;"
  >
    <input
      class="c2"
      placeholder="Добавить новый город"
    />
    <button
      class="c3"
      type="submit"
    >
      +
    </button>
  </form>
</div>
`;
